cocotb/share/lib/vpi/VpiCbHdl.cpp

```diff
@@ -49,77 +49,85 @@ VpiCbHdl::VpiCbHdl(GpiImplInterface *impl) : GpiCbHdl(impl)
     cb_data.user_data = (char*)this;
 }
 
-/* If the user data already has a callback handle then deregister
- * before getting the new one
- */
 int VpiCbHdl::arm_callback() {
+    switch(m_state) {
+    case GPI_PRIMED:
+    case GPI_CALL:
+        LOG_WARN("VPI: Attempted to prime an already primed trigger for %s (%d)",
+                  m_impl->reason_to_string(cb_data.reason), cb_data.reason);
+        /* FALLTHRU */
+    case GPI_FREE: {
+        if (m_obj_hdl != NULL) {
+            if (m_state == GPI_FREE) {
+                // LCOV_EXCL_START
+                LOG_ERROR("VPI: Simulator callback already registered for %s (%d)",
+                        m_impl->reason_to_string(cb_data.reason), cb_data.reason);
+                return -1;
+                // LCOV_EXCL_STOP
+            } else {
+                cleanup_callback();
```

Comment:
This won't cleanup the handle for `VpiTimedCbHdl`. Should we just return and do nothing if it's already primed?